### ISSUE TITLE
GuidanceComposer Missing Normalizer Jacobian

### ISSUE THREAD

**Issue by Max-Bin** [tier4/Diffusion-Planner#42]
## Summary

`GuidanceComposer.__call__()` computes guidance gradients in physical (denormalized) space but applies them as if they were in normalized space, missing a multiplication by `StateNormalizer.std`. This causes **position-related guidance (collision, lane-keeping, route-following) to be ~20x weaker than intended**.

`GuidanceWrapper` does NOT have this bug — it keeps the autograd graph intact through the inverse normalization, so the chain rule is applied automatically.

## Affected File

`diffusion_planner/diffusion_planner/model/guidance/composer.py` — `GuidanceComposer.__call__()` lines 50-78

## Root Cause

The composer detaches the trajectory before inverse-normalizing, breaking the autograd chain:

```python
# Line 50: detach breaks autograd graph
x_phys = state_normalizer.inverse(x_corrected.detach())

# Line 59: new autograd graph starts in physical space
x_phys_grad = x_phys.detach().requires_grad_(True)

# Lines 61-65: energy computed in physical space
raw_energy = ...
for fn in self._functions:
    e = fn.energy(x_phys_grad, t_scalar, inputs)
    raw_energy = raw_energy + e

# Line 68: gradient is dE/dx_phys (physical space)
grad_phys = torch.autograd.grad(raw_energy.sum(), x_phys_grad)[0]

# Line 74: applied directly as if it were dE/dx_norm (WRONG)
grad_flat = grad_phys.detach().reshape(x_in.shape)
surrogate = (grad_flat * x_in).sum()
```

The surrogate `dot(grad_phys, x_norm)` produces `autograd.grad(surrogate, x_norm) = grad_phys`, but the correct gradient should be `grad_phys * std` (by the chain rule: `dE/dx_norm = dE/dx_phys * dx_phys/dx_norm = dE/dx_phys * std`).

## Why GuidanceWrapper Is Correct

```python
# guidance_wrapper.py line 63: modifies x_in in-place (rebinds variable)
x_in = state_normalizer.inverse(x_in.reshape(B, P, -1, 4))

# Line 67: energy computed on physical x_in
energy += guidance_fn(x_in, t_input, cond, **kwargs)

# DPM solver later calls: torch.autograd.grad(energy, original_x_in)
# autograd correctly traces through inverse() (linear: x*std+mean)
# and returns dE/dx_phys * std automatically
```

## Numerical Proof

```python
import torch

std = torch.tensor([20.0, 20.0, 1.0, 1.0])
mean = torch.tensor([10.0, 0.0, 0.0, 0.0])
x_norm = torch.randn(1, 1, 5, 4, requires_grad=True)

# GuidanceWrapper (correct): autograd handles chain rule
x_phys_w = x_norm * std + mean
energy_w = (x_phys_w ** 2).sum()
grad_w = torch.autograd.grad(energy_w, x_norm)[0]

# GuidanceComposer (buggy): detach + surrogate misses std
x_phys_c = (x_norm * std + mean).detach().requires_grad_(True)
energy_c = (x_phys_c ** 2).sum()
grad_phys = torch.autograd.grad(energy_c, x_phys_c)[0]
surrogate = (grad_phys.detach() * x_norm).sum()
grad_c = torch.autograd.grad(surrogate, x_norm)[0]

ratio = grad_c / grad_w
# ratio = [0.05, 0.05, 1.0, 1.0] = 1/std
# Position gradients are 20x too small!
```

Output:
```
Ratio composer/wrapper: [0.05, 0.05, 1.0, 1.0]
Expected ratio (1/std): [0.05, 0.05, 1.0, 1.0]
```

## Impact

| Guidance Function | Affected Dimension | Scaling Error |
|---|---|---|
| CollisionGuidance | x, y position | 20x weaker |
| LaneKeepingGuidance | x, y position | 20x weaker |
| RouteFollowingGuidance | x, y position | 20x weaker |
| CenterlineFollowingGuidance | x, y position | 20x weaker |
| RoadBorderGuidance | x, y position | 20x weaker |
| All guidance | cos(yaw), sin(yaw) | Correct (std=1) |

## Affected Code Paths

| File | Line | Context |
|---|---|---|
| `preference_optimization/utils.py` | 291 | DPO trajectory sampling with guidance |
| `rlvr/grpo_sampler.py` | 198, 217, 330 | GRPO guided trajectory generation |
| `rlvr/grpo_exploration_trainer.py` | 246 | GRPO exploration training |
| `guidance_gui/generate_samples.py` | 24 | GUI debugging (cosmetic) |

**GuidanceWrapper is NOT affected** — it is used in the standard inference path and works correctly.

## Fix

Replace the surrogate approach with std-corrected gradients:

```python
# Option A: multiply by std before surrogate
grad_norm = grad_phys * state_normalizer.std.to(grad_phys.device)
grad_flat = grad_norm.detach().reshape(x_in.shape)
surrogate = (grad_flat * x_in).sum()

# Option B: keep autograd graph intact (like GuidanceWrapper)
# Remove .detach() from line 50, let autograd handle the chain rule
x_phys = state_normalizer.inverse(x_corrected)  # NO detach
```
